File: sheetsAPI.js
// sheetsAPI.js
const { google } = require('googleapis');

exports.getDataFromSheet = async function (cidade) {
    const auth = new google.auth.GoogleAuth({
        keyFile: 'credentials.json', // o caminho para suas credenciais
        scopes: 'https://www.googleapis.com/auth/spreadsheets.readonly',
    });

    const sheets = google.sheets({version: 'v4', auth});
    const spreadsheetId = '1FQOiNWM1ChoaLKK5a-26PvZDUo_iI993'; // Seu ID de Planilha
    const range = 'Sua_Aba!D:K'; // Ajuste a range conforme necessário

    try {
        const response = await sheets.spreadsheets.values.get({
            spreadsheetId: spreadsheetId,
            range: range,
        });
        const rows = response.data.values;
        const filteredRows = rows.filter(row => row[0].toLowerCase() === cidade.toLowerCase());
        return filteredRows;
    } catch (err) {
        console.error('Erro ao acessar o Google Sheets', err);
        throw err;
    }
};

File: sheets_api.py
# sheets_api.py
import gspread
from oauth2client.service_account import ServiceAccountCredentials

def get_data_from_sheet(city):
    # Defina o escopo de acesso ao Google Sheets
    scope = ['https://www.googleapis.com/auth/spreadsheets.readonly']

    # Carregue as credenciais do arquivo
    creds = ServiceAccountCredentials.from_json_keyfile_name('credentials.json', scope)
    client = gspread.authorize(creds)

    # Abra a planilha e a aba específica por ID e nome
    sheet = client.open_by_key('your_spreadsheet_id').worksheet('Nome_da_Aba')

    # Pega todos os registros da planilha
    data = sheet.get_all_records()

    # Filtra os dados pela cidade, considerando a correspondência de case
    filtered_data = [row for row in data if row['Cidade'].lower() == city.lower()]

    return filtered_data